onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1963,6 +2012,9 @@ PYBIND11_MODULE(onnxruntime_pybind11_state, m) {
   addGlobalMethods(m, env);
   addObjectMethods(m, env);
 
+  Ort::SessionOptions tmp_options;
+  OrtSessionOptionsAppendExecutionProvider_Dnnl(tmp_options, 0);
```

Comment:
Why DNNL?

Reply:
We need to initialize the Provider_host before any other shared library execution provider use it. After discuss with @RyanUnderhill , the easiest trick is to invoke this dnnl provider method once, it will complain dnnl provider not found but will do the initialization work. I agree it doesn't looks good. @RyanUnderhill , any suggestion?

Reply:
Only suggestion is to add a new API that only loads the shared library vs trying to also load a shared provider. That's cleaner, but has the drawback of exposing an API that only you will use. Not sure what is the lesser of the two evils.

Reply:
Oh yea, this is Python, just add a new API to load the shared provider, should be pretty simple.

Reply:
@RyanUnderhill we still need a C API for that.

Reply:
"We need to initialize the Provider_host before any other shared library execution provider use it". Why?  When you call ProviderSharedLibrary.Ensure(), it will manually load the shared library, but nobody would unload it.